mediapipe/examples/android/src/java/com/google/mediapipe/apps/instantmotiontracking/GIFEditText.java

```diff
@@ -39,43 +40,45 @@ public GIFEditText(Context context, AttributeSet attrs) {
+        this.gifCommitListener = gifCommitListener;
     }
 
     @Override
     public InputConnection onCreateInputConnection(EditorInfo editorInfo) {
-        final InputConnection inputConnection = super.onCreateInputConnection(editorInfo);
-        EditorInfoCompat.setContentMimeTypes(editorInfo, new String[]{"image/gif"});
-        return InputConnectionCompat.createWrapper(inputConnection, editorInfo,
-                new InputConnectionCompat.OnCommitContentListener() {
-                    @Override
-                    public boolean onCommitContent(final InputContentInfoCompat inputContentInfo,
-                                                   int flags, Bundle opts) {
-                            try {
-                                if (onGIFCommit != null) {
-                                    Runnable runnable = new Runnable() {
-                                        @Override
-                                        public void run() {
-                                            inputContentInfo.requestPermission();
-                                            onGIFCommit.OnGIFCommit(
-                                                    inputContentInfo.getContentUri(),
-                                                    inputContentInfo.getDescription());
-                                            inputContentInfo.releasePermission();
-                                        }
-                                    };
-                                    new Thread(runnable).start();
-                                }
-                            } catch (Exception e) {
-                                e.printStackTrace();
-                                return false;
-                            }
-                        return true;
-                    }
-                });
+      final InputConnection inputConnection = super.onCreateInputConnection(editorInfo);
+      EditorInfoCompat.setContentMimeTypes(editorInfo, new String[]{"image/gif"});
+      return InputConnectionCompat.createWrapper(inputConnection, editorInfo,
+          new InputConnectionCompat.OnCommitContentListener() {
+        @Override
+        public boolean onCommitContent(final InputContentInfoCompat inputContentInfo,
+            int flags, Bundle opts) {
+          try {
+            if (gifCommitListener != null) {
+              Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                  inputContentInfo.requestPermission();
+                  gifCommitListener.GIFCommitListener(
+                  inputContentInfo.getContentUri(),
+                  inputContentInfo.getDescription());
+                  inputContentInfo.releasePermission();
+                }
+              };
+              new Thread(runnable).start();
+            }
+          }
+          catch (Exception e) {
+            Log.e("GIFEditText", "Input connection to GIF selection failed");
```

Comment:
As a heads-up, if we have more logging, we'll probably want to make this a constant up at the top of the file-- but totally fine for now, since only one such.